runtime/queries/typescript/highlights.scm

```diff
@@ -22,15 +22,21 @@
 [
   "abstract"
   "declare"
-  "enum"
   "export"
   "implements"
-  "interface"
   "keyof"
   "namespace"
+] @keyword
+
+[
+  "type"
+  "interface"
+  "enum"
+] @keyword.storage.type
+
+[
+  "public"
   "private"
   "protected"
-  "public"
-  "type"
   "readonly"
-] @keyword
+] @keyword.storage.modifier
```

Comment:
These would probably be more appropriate as a new `keyword.storage.visibility` scope

Reply:
I'm not opposed to that for sure. I was just trying to follow the textmate grammar scopes as that's what's mentioned in the docs.

Are we happy to deviate from those?

Reply:
> Are we happy to deviate from those?

In general yes - we only use textmate (and sublime and vim) as a rough guide.

Textmate and sublime both have `public` as examples for `modifier` though so let's keep it under `keyword.storage.modifier` for now. We could introduce a `keyword.storage.modifier.visibility` later if it's needed.